lib/puppet/parser/functions/noop.rb

```diff
@@ -1,16 +1,16 @@
-# Set noop to true as default for current and children scopes
-Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default to true for all resources
+# Set noop to true as default for current and children scopes, or 'reset' an inherited default with noop(false)
+Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default for all resources
   in local scope and children scopes. This can be overriden in
   child scopes, or explicitly on each resource.
   ") do |args|
-
+      @noop_value = true if (@noop_value = args[0]).nil? or (![true, false].include? @noop_value)
```

Comment:
`(![true, false].include? @noop_value)` returns true when @noop_value is not either true or false. Yes a @noop_value of either true or false returns false there, so the `if` is not satisfied and the original value of @noop_value (true or false) is retained.
It's not as concise as double negation, but that's often considered bad style (e.g. Rubocop).
The tests now cover args nil, true, false and a string.